### ISSUE TITLE
FE | Profile | Address Validation Improvements

### ISSUE THREAD

**Issue by SKasimow** [department-of-veterans-affairs/va.gov-team#49652]
## Background

During recent research sessions, the address validation and editing flows were noted as being difficult to navigate for both screenreader users as well as non screenreader users. We've made some improvements to the designs which should result in a significant improvement tot he experience.

The desktop and mobile sketch files  are all available from the [use case documentation](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/profile/contact-information/use-cases/address-validation.md). There are 3 uses cases (6 screens in total).

Also it was noted that on staging, when the address can't be confirmed, and a suggestion is **not** found, we're showing the alert for when a suggestion is found, so FE will need to double check the logic on that too.

## Tasks
Following the mocks provided in the use case doc linked above, the changes are as follows:
- [x] update the secondary button label text
- [x] update the secondary button so it returns the user to edit mode instead of read mode
- [x] update the last sentence in all but 1 alert
- [x] When an address can't be confirmed and a suggestion is not available, we look to be showing the wrong alert (for when an address is found). This needs to be verified and updated.
- [x] Conduct Unit tests 
- [x] Run E2E tests covering current code and regression
- [x] Notify PM if this needs to be deployed to Staging behind feature flag
  

## Acceptance Criteria
- [x] Complete all unit testing
- [x] End-to-end tests showing 0 violations
- [x] All axe checks pass
- [x] Notify Product Manager that the change is ready to be validated on Staging
- [ ] Incorporate any changes resulting from Staging review & re-test & have re-validated
- [ ] Confirm PM has opened a new ticket to deploy changes to Production using feature flag, if needed

## Validation
- Who can validate this ticket? (FE, BE, Design, PM)?
- How can this work be validated?
- What updates need to be made (e.g. product outline, use cases, contact center guide)

## Associated Tickets
- #48537

## How to configure this issue
- Include practice area in title -- e.g., [Design], [FE], [BE]
- Include project/initiative name - e.g., Auth Profile: Address Change Messaging
- Add label for practice area (frontend, backend, design)
- Add label for project if applicable
- Assign to a Profile team member
- Associate with an Epic if applicable
- If creating for yourself: estimate work if enough info exists to do so

**Comment by Samara-Strauss** [comment]
Adam to review and then estimate.

**Comment by Samara-Strauss** [comment]
@andaleliz and @adamwhitlock1 -- can you review this together and confirm what needs to be done for the address validation changes? Feel free to update the requirements in the ticket as needed.

**Comment by andaleliz** [comment]
I just refreshed my memory of this work, and I think the tasks are accurate. Ultimately, we're solving the problem of "buttons that look like links" and cleaning up alerts. @adamwhitlock1 let me know if anything could use further clarification as you jump into this work.  [This comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/48537#issuecomment-1305777851) helped clear things up for me as I refreshed myself.

**Comment by Samara-Strauss** [comment]
Thanks, @andaleliz. Now that we're not in the middle of planning, I've looked at the screenshots and confirmed they are up-to-date via the link in the initial comment. Thank you for helping us all refresh our memories on this!

**Comment by adamwhitlock1** [comment]
Looking at the code and logic related around specifically displaying the alert at the top of the address validation UI, I have found a total of 9 validation types.

There are unique headings and text present for each scenario, although I think that some of these validation types _may_ never be reached in reality. I remember talking about how cases like `BAD_UNIT` and `BAD_UNIT_OVERRIDE` were really the same, or that one code never actually comes back from the address validation api responses.

In any case, I will be updating all of these, but wanted to make note of these cases for general education. We should probably make a ticket for the future to verify inactivity and remove the inactive cases/types to reduce tech debt and unused code bloat.

For reference, these are the messages that I am talking about

https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/user/profile/vap-svc/constants/addressValidationMessages.js

**Comment by Samara-Strauss** [comment]
Thanks, Adam. From the original project, I remember that we had initially considered not allowing override in some cases if the confidence score was low enough, and allowing override in other cases when the confidence score was high enough. However, we ultimately decided to allow override in **all** cases, so I assume anything that is not appended with `_OVERIDE` is not actually used. Sorry for the messy code. It looks like it was never cleaned up.

So, that should leave us with 5 potential use cases in reality:

- `badUnitNumberOverride`
- `missingUnitNumberOverride`
- `showSuggestionsOverride`
- `showSuggestionsNoConfirmedOverride`
- `validationError`

**Comment by adamwhitlock1** [comment]
Work is in review, and has been approved. Currently blocked by issues with the CI/CD pipelines, and specifically the unit tests. As soon as pipeline is passing, then the work will get deployed to staging for validation.

**Comment by adamwhitlock1** [comment]
![Screenshot 2023-02-14 at 5.29.27 PM.png](https://images.zenhubusercontent.com/60ddebd504ce2452d0e0fa92/33813fa9-d68d-424e-9f66-a80415766d9a)

Work has been merged, but currently not seeing it in staging. Hoping that by the morning the pr is deployed and we can validate.

**Comment by adamwhitlock1** [comment]
This is live in staging now @Samara-Strauss and @BerniXiongA6 and can be validated. A couple screenshots from when I tested things out with user 75

SUGGESTED ADDRESS FOUND
![Screenshot 2023-02-15 at 8 29 30 AM](https://user-images.githubusercontent.com/8332986/219083339-b9dda3d5-d77b-486f-8456-d480abf7706e.png)


NO UNIT NUMBER
![Screenshot 2023-02-15 at 8 33 54 AM](https://user-images.githubusercontent.com/8332986/219082905-59591b8f-ff2e-4d2a-b277-94fbfa02e871.png)

NO SUGGESTED ADDRESS FOUND
![Screenshot 2023-02-15 at 8 31 00 AM](https://user-images.githubusercontent.com/8332986/219083485-4387e382-dfd5-4d77-9d0e-c737579ec2b3.png)

Clicking the 'go back to edit' should now send the user back into the 'editing flow' with the form open for them to make edits instead of the old way of just closing the field altogether and discarding any edits that were made

**Comment by Samara-Strauss** [comment]
Validated with user 80! This work is good to go into prod.
